### ISSUE TITLE
Bug: Multiple assignments when using comma still breaks the rule `no-unused-expressions`.

### ISSUE THREAD

**Issue by IamFastre** [eslint/eslint#19039]
### Environment

Node version: v20.15.1
npm version: v10.8.2
Local ESLint version: v8.57.1 (Currently used)
Global ESLint version: v9.13.0
Operating System: win32 10.0.22621

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  "rules": {
    "no-unused-expressions": "error"
  }
}
```
</details>

```js
const directionText = "right";

const direction = {
  x: 0,
  y: 0,
};

if (directionText === 'up')
  direction.x = 0, direction.y = 1;
else if (directionText === 'right')
  direction.x = -1, direction.y = 0;
else if (directionText === 'down')
  direction.x = 0, direction.y = -1;
else if (directionText === 'left')
  direction.x = 1, direction.y = 0;
```

Or more minimally:

```js
let a = 0, b = 1;
a = 1, b = 2;
```

### What did you expect to happen?

No error should be fired 'cause the comma binary operation is of two assignments, meant to make the code more compact.

### What actually happened?

An unnecessary 'error' was fired.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgZGlyZWN0aW9uVGV4dCA9IFwicmlnaHRcIjtcblxuY29uc3QgZGlyZWN0aW9uID0ge1xuICB4OiAwLFxuICB5OiAwLFxufTtcblxuaWYgKGRpcmVjdGlvblRleHQgPT09ICd1cCcpXG4gIGRpcmVjdGlvbi54ID0gMCwgZGlyZWN0aW9uLnkgPSAxO1xuZWxzZSBpZiAoZGlyZWN0aW9uVGV4dCA9PT0gJ3JpZ2h0JylcbiAgZGlyZWN0aW9uLnggPSAtMSwgZGlyZWN0aW9uLnkgPSAwO1xuZWxzZSBpZiAoZGlyZWN0aW9uVGV4dCA9PT0gJ2Rvd24nKVxuICBkaXJlY3Rpb24ueCA9IDAsIGRpcmVjdGlvbi55ID0gLTE7XG5lbHNlIGlmIChkaXJlY3Rpb25UZXh0ID09PSAnbGVmdCcpXG4gIGRpcmVjdGlvbi54ID0gMSwgZGlyZWN0aW9uLnkgPSAwO1xuXG4iLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLXVudXNlZC1leHByZXNzaW9ucyI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

[More minimal example repro link](https://eslint.org/play/#eyJ0ZXh0IjoibGV0IGEgPSAwLCBiID0gMTtcbmEgPSAxLCBiID0gMjsiLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLXVudXNlZC1leHByZXNzaW9ucyI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=).

**Comment by mdjermanovic** [comment]
Hi @IamFastre, thanks for the issue!

This works as intended, per the [documentation](https://eslint.org/docs/latest/rules/no-unused-expressions) for this rule:

https://github.com/eslint/eslint/blob/725962731538eaa38d5d78b9e82ce3fccc9762d0/docs/src/rules/no-unused-expressions.md?plain=1#L29
